### ISSUE TITLE
More duplicate metric definitions for micrometer shim

### ISSUE THREAD

**Issue by tylerlance** [open-telemetry/opentelemetry-java#4510]
Hello.

This is a duplicate of [4381](https://github.com/open-telemetry/opentelemetry-java/issues/4381), but I wanted to bring this to your attention as this was assumed to be a rare scenario.

It appears the descriptions of Resilience4J meters are differentiated by descriptions.
````
[main] WARN io.opentelemetry.sdk.metrics.internal.state.MetricStorageRegistry - Found duplicate metric definition: resilience4j.circuitbreaker.calls
	at unknown source
		To enable better debugging, run your JVM with -Dotel.experimental.sdk.metrics.debug=true
Causes
- Description [Total number of failed calls] does not match [Total number of successful calls]
- InstrumentDescription [Total number of failed calls] does not match [Total number of successful calls]
Original instrument registered with same name but is incompatible.
````

Ideally we would want all these metrics sent to the collector, not just potentially the first one that was instrumented. Is the expectation going forward to update the descriptions in these other repos? Change the approach in this repo? Or should we be adding a MeterFilter to change the descriptions?

Current dependencies are:
micrometer-core v1.9.0
opentelemetry v1.14.0
resilience4j-spring-boot2 v1.7.1

**Comment by jack-berg** [comment]
Currently, all are sent to the collector. The warning you see is just a warning but both of the duplicate definitions are exported.

#4457 fixes this generally for micrometer metrics by caching the first seen description for a given metric name, and using that first description for subsequent instruments with the same name. This will be included in the upcoming 1.15.0 release.

**Comment by tylerlance** [comment]
Thanks @jack-berg. I'll close this issue and wait for the 1.15.0 release.
